catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -79,6 +78,35 @@ public MetacardTypeImpl(String name, Set<AttributeDescriptor> descriptors) {
         }
     }
 
+    /**
+     * Creates a {@code MetacardTypeImpl} with the provided {@link MetacardType}, set of
+     * additional {@linkplain AttributeDescriptor AttributeDescriptors}, and name.
+     * <p>
+     * {@code additionalDescriptors} and the descriptors in {@code metacardType} will be combined to
+     * form the set of descriptors for this {@code MetacardTypeImpl}.
+     * <p>
+     * Essentially, this is a convenience constructor for creating a new {@code MetacardTypeImpl}
+     * that extends an existing {@link MetacardType}.
+     *
+     * @param metacardType          the base {@link MetacardType}, cannot be null
+     * @param additionalDescriptors the descriptors to add to this {@code MetacardTypeImpl} in
+     *                              addition to the descriptors in {@code metacardType}, cannot be
+     *                              null or empty
+     * @param name                  the name of this {@code MetacardTypeImpl}
+     * @throws IllegalArgumentException if {@code metacardType} or {@code additionalDescriptors} is
+     *                                  null, or if {@code additionalDescriptors} is empty
+     */
+    public MetacardTypeImpl(MetacardType metacardType,
+            Set<AttributeDescriptor> additionalDescriptors, String name) {
```

Comment:
Why did you put the name last?  I would expect name to be first like in the other constructor.